ast/src/analyzed/mod.rs

```diff
@@ -370,6 +370,7 @@ pub struct PublicDeclaration {
     pub source: SourceRef,
     pub name: String,
     pub polynomial: PolynomialReference,
+    pub array_index: Option<DegreeType>,
```

Comment:
Why `DegreeType`? The array length isn't related to the size of the columns

Reply:
We essentially have "field element" or "degree type". Would you prefer `usize`?

Reply:
I think that'd be clearer